caproto/server/common.py

```diff
@@ -211,7 +212,14 @@ def get_db_entry():
                 command.data_type, user_address=self.circuit.address)
             # This is a pass-through if arr is None.
             data = apply_arr_filter(chan.channel_filter.arr, data)
-
+            # If the timestamp feature is active swap the timestamp.
+            # Information must copied because not all clients will have the
+            # timestamp filter
+            if chan.channel_filter.ts:
+                now = ca.TimeStamp.from_unix_timestamp(time.time())
+                metadata = DBR_TIME_LONG(status=metadata.status,
```

Comment:
I think we'd have to ensure that `command.data_type` is in `time_types` (https://github.com/NSLS-II/caproto/blob/master/caproto/_dbr.py#L1023), otherwise the request doesn't make sense. Then use `DBR_TYPES[type]` to get the appropriate `DBR_TYPE_*` structure.